### DISCUSSION TITLE
How to access *http.Request in templ component?

### DISCUSSION THREAD

**Discussion by sawirricardo** [a-h/templ#522] [Q&A]
We have context globally available in templ component. Can we achieve that with `req`?

**Comment by a-h** [comment]
Sorry, I don't understand the question.

**Reply by sawirricardo** [reply]
For instance
'''
// this is already working 👍
Templ My() {
<div>{ctx.Value("value").(string)
}

// can we do this?
Templ MyReq() {
<div>{req.URL.Path}</div>
}

**Reply by joerdav** [reply]
I'd recommend not doing so, parse your request and only pass into your components what is required. However, if this is needed for a niche scenario the your best bet would be to attach the request to the `ctx`, much like you added the string.
